src/script_opt/Inline.cc

```diff
@@ -120,33 +122,171 @@ void Inliner::Analyze() {
         inline_ables[func] = f.Profile();
     }
 
+    CoalesceEventHandlers();
+
     for ( auto& f : funcs )
-        if ( should_analyze(f.FuncPtr(), f.Body()) )
+        if ( f.ShouldAnalyze() )
             InlineFunction(&f);
 }
 
-void Inliner::InlineFunction(FuncInfo* f) {
-    max_inlined_frame_size = 0;
+void Inliner::CoalesceEventHandlers() {
+    std::unordered_map<ScriptFunc*, size_t> event_handlers;
+    BodyInfo body_to_info;
+    for ( size_t i = 0U; i < funcs.size(); ++i ) {
+        auto& f = funcs[i];
+        if ( ! f.ShouldAnalyze() )
+            continue;
+
+        auto& func_ptr = f.FuncPtr();
+        const auto& func = func_ptr.get();
+        const auto& func_type = func->GetType();
+
+        if ( func_type->AsFuncType()->Flavor() != FUNC_FLAVOR_EVENT )
+            continue;
+
+        // Special-case: zeek_init both has tons of event handlers (even
+        // with -b), such that it inevitably blows out the inlining budget,
+        // *and* only runs once, such that even if we could inline it, if
+        // it takes more time to compile it than to just run it via the
+        // interpreter, it's a lose.
+        static std::string zeek_init_name = "zeek_init";
+        if ( func->Name() == zeek_init_name )
+            continue;
+
+        const auto& body = f.Body();
+
+        if ( func->GetKind() == Func::SCRIPT_FUNC && func->GetBodies().size() > 1 ) {
+            if ( event_handlers.count(func) == 0 )
```

Comment:
I'll condense this to the following during the merge. `operator[]` will default construct an entry when missing.
```
++event_handlers[func];
```